mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/query/parsing/nativequery/impl/MongoDBQueryDescriptorBuilder.java

```diff
@@ -100,4 +152,37 @@ public DBObject parse(String json) {
 		return (DBObject) object.get( "json" );
 	}
 
+	private static DBObject stage(String key, Object value) {
+		DBObject stage = new BasicDBObject();
+		stage.put( key, value );
+		return stage;
+	}
+
+	public boolean addPipeline(StackElement element, String value) {
+		if ( !parsed.contains( element.getIndex() ) ) {
+			parsed.add( element.getIndex() );
+			boolean isJsonObject = value.trim().matches( "^\\{.*:.*\\}$" );
```

Comment:
Why don't you use the `parse` method directly for this? It was added exactly to deal with this and avoid this kind of tests.

It would avoid the regexp (which would have to be statically compiled anyway but let's get rid of it).